src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -195,7 +196,12 @@ private static async Task<HttpResponseMessage> SendWithNtAuthAsync(HttpRequestMe
                                 if (!IsAuthenticationChallenge(response, isProxyAuth))
                                 {
                                     // Tail response for Negoatiate on successful authentication. Validate it before we proceed.
-                                    authContext.GetOutgoingBlob(challengeData);
+                                    authContext.GetOutgoingBlob(challengeData, throwOnError: false, out statusCode);
+                                    if (statusCode.ErrorCode != SecurityStatusPalErrorCode.OK)
+                                    {
+                                        await connection.DrainResponseAsync(response!, cancellationToken).ConfigureAwait(false);
```

Comment:
What are the conditions we can get here? I'm wondering if we should close the connection to prevent re-use in some weird state.

Reply:
Message has been altered and fails the last integrity check. This ensures that both parties in the Negotiate authentication were negotiating over the same protocol set, ie. prevents MITM protocol downgrade attack.

Reply:
So this should be rare, right? (and possibly sign of something malicious)
It feels like closing the connection would be better choice. (we normally drain so the connection could be re-used)